src/EnergyPlus/CMakeLists.txt

```diff
@@ -668,8 +680,24 @@ add_custom_target( GenerateEmbeddedEpJSONSchema
                    DEPENDS generate_embeddedable_epJSON_schema )
 
 add_subdirectory(AirflowNetwork)
-#message( $<TARGET_PROPERTY:airflownetworklib,INTERFACE_INCLUDE_DIRECTORIES> )
-#include_directories($<TARGET_PROPERTY:airflownetworklib,INTERFACE_INCLUDE_DIRECTORIES>)
+
+if(LINK_WITH_PYTHON)
+  # check the python bit version here...if we are building against Python it has to match up bitwise
+  # if we aren't, then the identified interpreter can be whatever bit it wants
+  if(MSVC)
+    execute_process(COMMAND "${PYTHON_EXECUTABLE}" "${CMAKE_SOURCE_DIR}/cmake/PythonGetBitSize.py" OUTPUT_VARIABLE PYTHON_BIT_SIZE)
+    if(("${PYTHON_BIT_SIZE}" STREQUAL "64") AND NOT ("${CMAKE_GENERATOR_PLATFORM}" STREQUAL "Win64"))
+      message(SEND_ERROR "Found 64-bit Python but attempting to generate 32-bit EnergyPlus; this is invalid")
+    endif()
+    if(("${PYTHON_BIT_SIZE}" STREQUAL "32") AND ("${CMAKE_GENERATOR_PLATFORM}" STREQUAL "Win64"))
+      message(SEND_ERROR "Found 32-bit Python but attempting to generate 64-bit EnergyPlus; this is invalid")
+    endif()
```

Comment:
@Myoldmopar I have 4 choices as the platform on MSVS. Do all those need to be accounted for (e.g., if ARM, error)? Also, I have x64, is that translated to "Win64" in CMake? And shouldn't the 2nd one also be AND NOT "Win32" so a mixture of other possibilities don't confuse the logic?

![image](https://user-images.githubusercontent.com/5778156/74970741-e3f39480-53ec-11ea-85aa-8b5dd1832fec.png)

Reply:
We don't need ARM to be accounted.  Your question on the Win64 v Win32 is valid, I'll verify that real quick.

Reply:
OH!  It's not Win64 then.  It's either Win32 or x64 ... https://cmake.org/cmake/help/latest/generator/Visual%20Studio%2016%202019.html

I'll push that change up, and try to clean up the logic in that section.  I think that should fix the error handling at least.

@EnergyArchmage can you confirm whether you can run a PythonPlugin example file in any/all of those builds you made?

Reply:
Yeah, or not...I just did some debugging and found that the CMAKE_GENERATOR_PLATFORM variable may or may not be set, depending on CMake version and generator.  Gross.  It's going to take a little more thought to get that error handling in place, but I'll get there.  For now I just had to comment the errors -- they weren't doing any good anyway.